vendor/bat-native-ledger/src/bat/ledger/internal/legacy/publisher_settings_properties.cc

```diff
@@ -48,4 +72,147 @@ bool PublisherSettingsProperties::operator!=(
+  dict.Set(kMinVisitsForPublisherRelevancy,
+           base::Int64ToValue(min_visits_for_publisher_relevancy));
+  dict.Set(kAllowNonVerifiedSitesInListKey, allow_non_verified_sites_in_list);
+  dict.Set(kAllowContributionToVideosKey, allow_contribution_to_videos);
+
+  base::Value::List monthly_balances_list;
+  for (const auto& monthly_balance : monthly_balances) {
+    base::Value::Dict item;
+    item.Set(monthly_balance.first, monthly_balance.second.ToValue());
+    monthly_balances_list.Append(std::move(item));
+  }
+  dict.Set(kMonthlyBalancesKey, std::move(monthly_balances_list));
+
+  base::Value::List prcessed_pending_publishers_list;
+  for (const auto& processed_pending_publisher : processed_pending_publishers) {
+    prcessed_pending_publishers_list.Append(processed_pending_publisher);
+  }
+  dict.Set(kProcessedPendingPublishersKey,
+           std::move(prcessed_pending_publishers_list));
+
+  return dict;
+}
+
+bool PublisherSettingsProperties::FromValue(const base::Value::Dict& dict) {
+  // Min Publisher Duration (There is no support for uint64_t. Writing JSON with
+  // such types violates the spec. As we need a uint64_t, we need to use a
+  // double and cast to a uint64_t)
+  if (auto value =
+          base::ValueToInt64(dict.Find(kMinPageTimeBeforeLoggingAVisitKey))) {
+    min_page_time_before_logging_a_visit = *value;
+  } else if (auto value = dict.FindDouble(kMinPageTimeBeforeLoggingAVisitKey)) {
+    min_page_time_before_logging_a_visit = static_cast<uint64_t>(*value);
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Min Visits (There is no support for unsigned int. Writing JSON with such
+  // types violates the spec. As we need an unsigned int, we need to a double
+  // and cast to an unsigned int)
+  if (auto value =
+          base::ValueToInt64(dict.Find(kMinVisitsForPublisherRelevancy))) {
+    min_visits_for_publisher_relevancy = *value;
+  } else if (auto value = dict.FindDouble(kMinVisitsForPublisherRelevancy)) {
+    min_visits_for_publisher_relevancy = static_cast<unsigned int>(*value);
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Allow Non Verified
+  if (auto value = dict.FindBool(kAllowNonVerifiedSitesInListKey)) {
+    allow_non_verified_sites_in_list = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Allow Videos
+  if (auto value = dict.FindBool(kAllowContributionToVideosKey)) {
+    allow_contribution_to_videos = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Monthly Balances
+  const auto* monthly_balances_list = dict.FindList(kMonthlyBalancesKey);
+  if (!monthly_balances_list) {
+    NOTREACHED();
+    return false;
+  }
+
+  for (const auto& item : *monthly_balances_list) {
+    const auto* monthly_balance_value = item.GetIfDict();
+    if (!monthly_balance_value) {
+      NOTREACHED();
+      continue;
+    }
+
+    for (const auto item : *monthly_balance_value) {
```

Comment:
Structured binding.

Reply:
Fixed in 0338b1ef458d495a3d9188cf7ca28128f2b9c85c.